MobileWallet/Common/NotificationManager.swift

```diff
@@ -95,4 +155,220 @@ class NotificationManager {
+
+        let task = URLSession.shared.dataTask(with: request) { data, response, error in
+            guard error == nil else {
+                onRequestError(error!)
+                return
+            }
+
+            guard let data = data, let response = response as? HTTPURLResponse else {
+                onRequestError(TokenServerError.unknown)
+                return
+            }
+
+            guard response.statusCode != 403 else {
+                onRequestError(TokenServerError.invalidSignature)
+                return
+            }
+
+            var responseDict: [String: Any]?
+            if let responseString = String(data: data, encoding: .utf8) {
+                if let data = responseString.data(using: .utf8) {
+                    do {
+                        responseDict = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any]
+                    } catch {
+                        onRequestError(error)
+                        return
+                    }
+                }
+            }
+
+            guard (200 ... 299) ~= response.statusCode else {
+                var message: String?
+                if let res = responseDict {
+                    message = res["error"] as? String
+                }
+
+                onRequestError(TokenServerError.server(response.statusCode, message: message))
+                return
+            }
+
+            guard let res = responseDict else {
+                onRequestError(TokenServerError.missingResponse)
+                return
+            }
+
+            guard let registered = res["registered"] as? Bool else {
+                onRequestError(TokenServerError.responseInvalid)
+                return
+            }
+
+            TariLogger.info("Device token reigistered with public key")
+            UserDefaults.standard.set(true, forKey: NotificationManager.HAS_REGISTERED_TOKEN_STORAGE_KEY)
+
+            NotificationManager.isRegisterRequestInProgress = false
+        }
+
+        task.resume()
+    }
+
+    func sendToRecipient(_ toPublicKey: PublicKey, onSuccess: @escaping (() -> Void), onError: @escaping ((Error) -> Void)) throws {
+        guard let wallet = TariLib.shared.tariWallet else {
+            return TariLogger.error("Failed to get wallet", error: WalletErrors.walletNotInitialized)
+        }
+
+        let (pubKey, pubKeyError) = wallet.publicKey
+        guard pubKeyError == nil else {
+            return TariLogger.error("Failed to get wallet pub key", error: pubKeyError)
+        }
+
+        let (fromPubKeyHex, hexError) = pubKey!.hex
+        guard hexError == nil else {
+            return TariLogger.error("Failed to get wallet hex pub key", error: hexError)
+        }
+
+        let (toPubKeyHex, toHexError) = toPublicKey.hex
+        guard toHexError == nil else {
+            return TariLogger.error("Failed to get wallet to hex pub key", error: hexError)
+        }
```

Comment:
Same `guard` series.

Reply:
What's the best way to avoid this?

Reply:
Specific to these lines (we can come back to these later, just as a note): wallet returning the public key right is a unit test check, and any public key that was constructed successfully should return a hex string right (i think).